mi/dataset/driver/zplsc_c/zplsc_functions.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+"""
+@package mi.dataset.driver.zplsc_c.zplsc_functions
+@file mi.dataset.driver.zplsc_c/zplsc_functions.py
+@author Rene Gelinas
+@brief Module containing ZPLSC related data calculations.
+"""
+
+import numpy as np
+
+
+def zplsc_b_decompress(power):
+    """
+    Description:
+
+        Convert a list of compressed power values to numpy array of
+        decompressed power.  This code was from the zplsc_b.py parser,
+        when it only produced an echogram.
+
+    Implemented by:
+
+        2017-06-27: Rene Gelinas. Initial code.
+
+    :param power: List of zplsc B series power values in a compressed format.
+    :return: Decompressed numpy array of power values.
+    """
+
+    decompress_power = np.array(power) * 10. * np.log10(2) / 256.
+
+    return decompress_power
+
+
+def zplsc_c_temperature(counts, ka, kb, kc, a, b, c):
+    """
+    Description:
+
+        Compute the temperature from the counts passed in.
+        This Code was lifted from the ASL MatLab code LoadAZFP.m
+
+    Implemented by:
+
+        2017-06-23: Rene Gelinas. Initial code.
+
+    :param counts: Raw data temperature counts from the zplsc-c raw data file.
+    :param ka:
+    :param kb:
+    :param kc:
+    :param a:
+    :param b:
+    :param c:
+    :return: temperature
+    """
+
+    vin = 2.5 * (counts / 65535)
+    r = (ka + kb*vin) / (kc - vin)
+    temperature = 1 / (a + b * (np.log(r)) + c * (np.log(r)**3)) - 273
+
+    return temperature
+
+
+def zplsc_c_tilt(counts, a, b, c, d):
+    """
+    Description:
+
+        Compute the tilt from the counts passed in from the zplsc A series.
+        This Code was from the ASL MatLab code LoadAZFP.m
+
+    Implemented by:
+
+        2017-06-23: Rene Gelinas. Initial code.
+
+    :param counts:
+    :param a:
+    :param b:
+    :param c:
+    :param d:
+    :return: tilt value
+    """
+
+    tilt = a + (b * counts) + (c * counts**2) + (d * counts**3)
+
+    return tilt
+
+
+def zplsc_c_ss(t, p, s):
+    """
+    Description:
+
+        Compute the ss from the counts passed in.
+        This Code was from the ASL MatLab code LoadAZFP.m
+
+    Implemented by:
+
+        2017-06-23: Rene Gelinas. Initial code.
+
+    :param t:
+    :param p:
+    :param s:
+    :return:
+    """
+
+    z = t/10
+    sea_c = 1449.05 + (z * (45.7 + z*((-5.21) + 0.23*z))) + ((1.333 + z*((-0.126) + z*0.009)) * (s-35.0)) + \
+        (p/1000)*(16.3+0.18*(p/1000))
+
+    return sea_c
+
+
+def zplsc_c_absorbtion(t, p, s, freq):
+    """
+    Description:
+
+        Calculate Absorption coeff using Temperature, Pressure and Salinity and transducer frequency.
+        This Code was from the ASL MatLab code LoadAZFP.m
+
+    Implemented by:
+
+        2017-06-23: Rene Gelinas. Initial code.
+
+
+    :param t:

```

Comment:
Add name and units for each parameter if available

Reply:
The code in this file is reused code from the ASL MatLab ZPLSC Echogram code.  The ASL code had virtually no comments.  Many of the variables are single character variables without explanation.